rxjava/src/main/java/com/linecorp/armeria/common/rxjava/RequestContextSupplierFlowable.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.rxjava;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.util.SafeCloseable;
+
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.functions.Supplier;
+import io.reactivex.rxjava3.internal.fuseable.ConditionalSubscriber;
+
+final class RequestContextSupplierFlowable<T> extends Flowable<T> implements Supplier<T> {
+    private final Publisher<T> source;
+    private final RequestContext assemblyContext;
+
+    RequestContextSupplierFlowable(Publisher<T> source, RequestContext assemblyContext) {
+        this.source = source;
+        this.assemblyContext = assemblyContext;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected void subscribeActual(Subscriber<? super T> s) {
+        try (SafeCloseable ignored = assemblyContext.push()) {
+            if (s instanceof ConditionalSubscriber) {
+                source.subscribe(new RequestContextConditionalSubscriber<>((ConditionalSubscriber<? super T>) s,
+                                                                           assemblyContext));
+            } else {
+                source.subscribe(new RequestContextSubscriber<>(s, assemblyContext));
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public T get() throws Throwable {
+        try (SafeCloseable ignored = assemblyContext.push()) {
```

Comment:
As the comment above, do we need to remove here too?

Reply:
This implements `Supplier<T>`, not `ScalarSupplier<T>`. So I think this is fine. 😀